pkg/operator/ceph/cluster/mgr/spec.go

```diff
@@ -39,22 +40,46 @@ func (c *Cluster) makeDeployment(mgrConfig *mgrConfig) *apps.Deployment {
 			Labels: c.getPodLabels(mgrConfig.DaemonID),
 		},
 		Spec: v1.PodSpec{
-			InitContainers: []v1.Container{
-				c.makeSetServerAddrInitContainer(mgrConfig, "dashboard"),
-				c.makeSetServerAddrInitContainer(mgrConfig, "prometheus"),
-			},
+			InitContainers: []v1.Container{},
 			Containers: []v1.Container{
 				c.makeMgrDaemonContainer(mgrConfig),
 			},
 			ServiceAccountName: serviceAccountName,
 			RestartPolicy:      v1.RestartPolicyAlways,
-			Volumes: append(
-				opspec.DaemonVolumes(mgrConfig.DataPathMap, mgrConfig.ResourceName),
-				keyring.Volume().Admin(), // ceph config set commands want admin keyring
-			),
-			HostNetwork: c.HostNetwork,
+			Volumes:            opspec.DaemonVolumes(mgrConfig.DataPathMap, mgrConfig.ResourceName),
+			HostNetwork:        c.HostNetwork,
 		},
 	}
+
+	needHttpBindFix := true
+
+	// if luminous and >= 12.2.12
+	if c.clusterInfo.CephVersion.IsLuminous() &&
+		c.clusterInfo.CephVersion.IsAtLeast(cephver.CephVersion{Major: 12, Minor: 2, Extra: 12}) {
+		needHttpBindFix = false
+	}
+
+	// if mimic and >= 13.2.6
+	if c.clusterInfo.CephVersion.IsMimic() &&
+		c.clusterInfo.CephVersion.IsAtLeast(cephver.CephVersion{Major: 13, Minor: 2, Extra: 6}) {
+		needHttpBindFix = false
+	}
+
+	// if >= 14.1.1
+	if c.clusterInfo.CephVersion.IsAtLeast(cephver.CephVersion{Major: 14, Minor: 1, Extra: 1}) {
```

Comment:
Is there a simple update to the unit tests in spec_test.go that could cover at least a couple version permutations of this change?